packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx

```diff
@@ -285,6 +294,46 @@ export let DataSpreadsheet = React.forwardRef(
       [cellEditorValue, onDataUpdate]
     );
 
+    useEffect(() => {
+      const currentHeaders: Array<any> = [];
+      const pastHeaders: Array<any> = [];
+      if (Object.keys(currentColumns).length > 0) {
+        Object.keys(currentColumns).forEach((itemIndex) => {
+          if (currentColumns[itemIndex].Header) {
+            currentHeaders.push(currentColumns[itemIndex].Header);
+          }
+        });
+      }
+      if (Object.keys(columns).length > 0) {
+        Object.keys(columns).forEach((itemIndex) => {
+          if (columns[itemIndex].Header) {
+            pastHeaders.push(columns[itemIndex].Header);
+          }
+        });
+      }
+
+      if (
+        !previousState.selectedHeaderReorderActive &&
+        !headerCellHoldActive &&
+        currentHeaders.length > 0 &&
+        pastHeaders.length > 0 &&
+        JSON.stringify(currentHeaders) !== JSON.stringify(pastHeaders)
+      ) {
+        // Return back data
+        onColDrag({
+          headers: currentHeaders,
+          data: activeCellContent.props.data,
+        });
+      }
+    }, [
+      previousState?.selectedHeaderReorderActive,
+      currentColumns,
+      headerCellHoldActive,
+      columns,
+      activeCellContent,
+      onColDrag,
+    ]);
+
```

Comment:
Hey @AustinGitHub ,
Code looks fine, but I was thinking of this scenario. 
Let initial order of columns be `orderDefault`. 
When colums are dragged , order changed from `orderDefault` to `order 1` , then changed to `order 2`. Then again dragged back to `orderDefault`. At this last drag scenario , will this callback get triggers?
@davidmenendez Can you share your thoughts? Is this a required scenario ?

Reply:
Let me check

Reply:
You're right, it doesn't get triggered in this use case. I will add a fix for this